torch/onnx/symbolic_opset17.py

```diff
@@ -54,3 +55,128 @@ def layer_norm(
         epsilon_f=eps,
         axis_i=axis,
     )
+
+
+@_onnx_symbolic("aten::stft")
+@symbolic_helper.parse_args("v", "i", "i", "i", "v", "b", "b", "b")
+@_beartype.beartype
+def stft(
+    g: jit_utils.GraphContext,
+    input: _C.Value,
+    n_fft: int,
+    hop_length: Optional[int] = None,
+    win_length: Optional[int] = None,
+    window: Optional[_C.Value] = None,
+    normalized: bool = False,
+    onesided: Optional[bool] = True,
+    return_complex: Optional[bool] = False,
+) -> _C.Value:
+    """Associates `torch.stft` with the `STFT` ONNX operator.
+    Note that torch.stft calls _VF.stft, without centering or padding options.
+    Hence, this function does not contain these two arguments.
+    See torch.stft source code for more info.
+
+    Parameters
+    ----------
+    g : jit_utils.GraphContext
+        Graph to write the ONNX representation into
+    input : _C.Value
+        Input tensor for the transformation
+    n_fft : int
+        FFT size
+    hop_length : int, optional
+        Size of the hop. Defaults to floot(n_fft // 4)
+    win_length : int, optional
+        Size of the analysis window. Defaults to `n_fft`
+    window : _C.Value, optional
+        Analysis window. Defaults to a window of all ones.
+    normalized : bool, optional
+        Whether to return a normalized STFT.
+    onesided : bool, optional
+        Whether to return only half (+1) of the results, given the symmetry
+        of the STFT
+    return_complex : bool, optional
+        Whether to return the complex value (Note: Must be False or None)
+
+    Returns
+    -------
+    op
+        Operator for torch.stft associated with STFT (ONNX).
+    """
+    # Checks
+    assert (
+        return_complex is None or not return_complex
+    ), "STFT does not currently support complex types"
```

Comment:
`return_complex=False` is deprecated so I don't like the idea of forcing people to use it. I see that complex64 and complex128 are mentioned as types in [this document](https://github.com/onnx/onnx/blob/main/docs/Operators.md), so is it possible to implement `view_as_real`/`view_as_complex` in onnx?

Reply:
I like the idea of having view_as_real and view_as_complex ops in onnx. Would you be willing to open an issue for a new operator? https://github.com/onnx/onnx/issues/new?assignees=&labels=operator&template=operator.md&title=

Reply:
@peterbell10 I also don't like the idea of forcing `return_complex=False`, but it is an easy workaround if we don't want to wait to have complex conversion support on ONNX, which seems like it might take quite some time (see below).

@justinchuby Seems like `view_as_complex` was already requested here: https://github.com/pytorch/pytorch/issues/49793 Unfortunately, there was not enough interest for this when it was reported. I just commented on the issue, to try to bring it back.

Reply:
Am I right in saying onnx Cast doesn't support complex types? If it did then you could do something like:
```python
def _as_real(z):
    return float(z), float(-1j * z)

def _as_complex(real, imag):
    return complex(real) + 1j * complex(imag)
```

Reply:
That is correct: ONNX doesn't currently support casting for complex types (https://github.com/onnx/onnx/blob/main/docs/Operators.md#Cast).

If it did, we could simply use `Cast` on the result of `STFT` to make sure the result is returned as complex if `return_complex=True`.

That being said, would it be possible to obtain the output of `STFT`, put it on a PyTorch tensor, and convert it to complex using the `_as_complex` function above, and then put it back into the ONNX Graph?

Reply:
Yeah, otherwise there will be a graph break in onnx.